homeassistant/components/geo_location/usgs_earthquake_hazards_program_feed.py

```diff
@@ -0,0 +1,269 @@
+"""
+U.S. Geological Survey Earthquake Hazards Program Feed platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/geo_location/
+usgs_earthquake_hazards_program_feed/
+"""
+from datetime import timedelta
+import logging
+from typing import Optional
+
+import voluptuous as vol
+
+from homeassistant.components.geo_location import (
+    PLATFORM_SCHEMA, GeoLocationEvent)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_RADIUS, CONF_SCAN_INTERVAL,
+    EVENT_HOMEASSISTANT_START, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['geojson_client==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ALERT = 'alert'
+ATTR_EXTERNAL_ID = 'external_id'
+ATTR_MAGNITUDE = 'magnitude'
+ATTR_PLACE = 'place'
+ATTR_STATUS = 'status'
+ATTR_TIME = 'time'
+ATTR_TYPE = 'type'
+ATTR_UPDATED = 'updated'
+
+CONF_FEED_TYPE = 'feed_type'
+CONF_MINIMUM_MAGNITUDE = 'minimum_magnitude'
+
+DEFAULT_MINIMUM_MAGNITUDE = 0.0
+DEFAULT_RADIUS_IN_KM = 50.0
+DEFAULT_UNIT_OF_MEASUREMENT = 'km'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+SIGNAL_DELETE_ENTITY = 'usgs_earthquake_hazards_program_feed_delete_{}'
+SIGNAL_UPDATE_ENTITY = 'usgs_earthquake_hazards_program_feed_update_{}'
+
+SOURCE = 'usgs_earthquake_hazards_program_feed'
+
+VALID_FEED_TYPES = [
+    'past_hour_significant_earthquakes',
+    'past_hour_m45_earthquakes',
+    'past_hour_m25_earthquakes',
+    'past_hour_m10_earthquakes',
+    'past_hour_all_earthquakes',
+    'past_day_significant_earthquakes',
+    'past_day_m45_earthquakes',
+    'past_day_m25_earthquakes',
+    'past_day_m10_earthquakes',
+    'past_day_all_earthquakes',
+    'past_week_significant_earthquakes',
+    'past_week_m45_earthquakes',
+    'past_week_m25_earthquakes',
+    'past_week_m10_earthquakes',
+    'past_week_all_earthquakes',
+    'past_month_significant_earthquakes',
+    'past_month_m45_earthquakes',
+    'past_month_m25_earthquakes',
+    'past_month_m10_earthquakes',
+    'past_month_all_earthquakes',
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FEED_TYPE): vol.In(VALID_FEED_TYPES),
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM): vol.Coerce(float),
+    vol.Optional(CONF_MINIMUM_MAGNITUDE, default=DEFAULT_MINIMUM_MAGNITUDE):
+        vol.All(vol.Coerce(float), vol.Range(min=0))
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the USGS Earthquake Hazards Program Feed platform."""
+    scan_interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    feed_type = config.get(CONF_FEED_TYPE)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.

Reply:
OK